apps/meteor/server/services/omnichannel-verification/service.ts

```diff
@@ -0,0 +1,166 @@
+import type { IRoom, IOmnichannelGenericRoom } from '@rocket.chat/core-typings';
+import { VerificationStatusEnum, RoomVerificationState } from '@rocket.chat/core-typings';
+import { check } from 'meteor/check';
+import type { IOmnichannelVerification } from '@rocket.chat/core-services';
+import { ServiceClassInternal } from '@rocket.chat/core-services';
+import { LivechatVisitors, LivechatRooms, Users } from '@rocket.chat/models';
+import { Random } from '@rocket.chat/random';
+import { Accounts } from 'meteor/accounts-base';
+import bcrypt from 'bcrypt';
+
+import { sendMessage } from '../../../app/lib/server/functions/sendMessage';
+import { i18n } from '../../lib/i18n';
+import { settings } from '../../../app/settings/server';
+import { Logger } from '../../../app/logger/server';
+import * as Mailer from '../../../app/mailer/server/api';
+
+export class OmnichannelVerification extends ServiceClassInternal implements IOmnichannelVerification {
+	protected name = 'omnichannel-verification';
+
+	private logger: Logger;
+
+	constructor() {
+		super();
+		this.logger = new Logger('GetVerficationCode');
+	}
+
+	private async send2FAEmail(address: string, random: string): Promise<void> {
+		const language = settings.get<string>('Language') || 'en';
+
+		const t = (s: string): string => i18n.t(s, { lng: language });
+		await Mailer.send({
+			to: address,
+			from: settings.get('From_Email'),
+			subject: 'Authentication code',
+			replyTo: undefined,
+			data: {
+				code: random.replace(/^(\d{3})/, '$1-'),
+			},
+			headers: undefined,
+			text: `
+    ${t('Here_is_your_authentication_code')}
+    
+    __code__
+    
+    ${t('Do_not_provide_this_code_to_anyone')}
+    ${t('If_you_didnt_try_to_login_in_your_account_please_ignore_this_email')}
+    `,
+			html: `
+                <p>${t('Here_is_your_authentication_code')}</p>
+                <p style="font-size: 30px;">
+                    <b>__code__</b>
+                </p>
+                <p>${t('Do_not_provide_this_code_to_anyone')}</p>
+                <p>${t('If_you_didnt_try_to_login_in_your_account_please_ignore_this_email')}</p>
+            `,
+		});
+	}
+
+	private async sendVerificationCodeToVisitor(visitorId: string): Promise<void> {
+		if (!visitorId) {
+			throw new Error('error-invalid-user');
+		}
+		const visitor = await LivechatVisitors.findOneById(visitorId, {});
+
+		if (!visitor) {
+			throw new Error('error-invalid-user');
+		}
+
+		if (!visitor?.visitorEmails?.length) {
+			throw new Error('error-email-required');
+		}
+		const visitorEmail = visitor.visitorEmails[0].address;
+		const random = Random._randomString(6, '0123456789');
+		const encryptedRandom = await bcrypt.hash(random, Accounts._bcryptRounds());
+		const expire = new Date();
+		const expirationInSeconds = parseInt(settings.get('Accounts_TwoFactorAuthentication_By_Email_Code_Expiration') as string, 10);
+		expire.setSeconds(expire.getSeconds() + expirationInSeconds);
```

Comment:
We can move the logic to generate OTP and expiration time to a helper function and make this function more readable